src/main/java/net/mcreator/ui/modgui/VillagerTradeGUI.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2021, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.modgui;
+
+import net.mcreator.element.types.VillagerTrade;
+import net.mcreator.ui.MCreator;
+import net.mcreator.ui.component.util.PanelUtils;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.init.UIRES;
+import net.mcreator.ui.minecraft.villagers.JVillagerTradeProfession;
+import net.mcreator.ui.validation.AggregatedValidationResult;
+import net.mcreator.workspace.elements.ModElement;
+
+import javax.annotation.Nonnull;
+import javax.swing.*;
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class VillagerTradeGUI extends ModElementGUI<VillagerTrade> {
+
+	private final List<JVillagerTradeProfession> professionList = new ArrayList<>();
+
+	private final JPanel professions = new JPanel(new GridLayout(0, 1, 5, 5));
```

Comment:
This should also be migrated to `JEntriesList` like it was for loot tables (not the GUI itself, check #2246 for reference), as well as professions should be registered in it.

Reply:
Yeah but Loot Tables can be used in datapacks, but Villager Trades can't, they are exclusive to mods

Reply:
Nevermind, I found a way and it's done.